File: app/src/main/java/com/rta/vtc/PrefsManager.java
package com.rta.vtc;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class PrefsManager {
    private static final String PREFS_NAME = "rta_vtc_prefs";
    private static final String KEY_DISPLAY_NAME = "display_name";
    private static final String KEY_MEETING_HISTORY = "meeting_history";
    private static final String KEY_AUDIO_MUTED = "audio_muted";
    private static final String KEY_VIDEO_MUTED = "video_muted";
    private static final int MAX_HISTORY = 10;

    private final SharedPreferences prefs;
    private final Gson gson;

    public PrefsManager(Context context) {
        prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        gson = new Gson();
    }

    public void setDisplayName(String name) {
        prefs.edit().putString(KEY_DISPLAY_NAME, name).apply();
    }

    public String getDisplayName() {
        return prefs.getString(KEY_DISPLAY_NAME, "");
    }

    public void setAudioMuted(boolean muted) {
        prefs.edit().putBoolean(KEY_AUDIO_MUTED, muted).apply();
    }

    public boolean isAudioMuted() {
        return prefs.getBoolean(KEY_AUDIO_MUTED, false);
    }

    public void setVideoMuted(boolean muted) {
        prefs.edit().putBoolean(KEY_VIDEO_MUTED, muted).apply();
    }

    public boolean isVideoMuted() {
        return prefs.getBoolean(KEY_VIDEO_MUTED, false);
    }

    public void addMeetingToHistory(String roomName) {
        List<MeetingItem> history = getMeetingHistory();
        // Remove if already exists
        history.removeIf(item -> item.roomName.equals(roomName));
        // Add to top
        history.add(0, new MeetingItem(roomName, System.currentTimeMillis()));
        // Limit size
        if (history.size() > MAX_HISTORY) {
            history = new ArrayList<>(history.subList(0, MAX_HISTORY));
        }
        prefs.edit().putString(KEY_MEETING_HISTORY, gson.toJson(history)).apply();
    }

    public List<MeetingItem> getMeetingHistory() {
        String json = prefs.getString(KEY_MEETING_HISTORY, "[]");
        Type type = new TypeToken<ArrayList<MeetingItem>>() {}.getType();
        List<MeetingItem> list = gson.fromJson(json, type);
        return list != null ? new ArrayList<>(list) : new ArrayList<>();
    }

    public void clearHistory() {
        prefs.edit().remove(KEY_MEETING_HISTORY).apply();
    }

    public static class MeetingItem {
        public String roomName;
        public long timestamp;

        public MeetingItem(String roomName, long timestamp) {
            this.roomName = roomName;
            this.timestamp = timestamp;
        }
    }
}

File: rta_vtc_flutter/lib/services/prefs_manager.dart
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';

class MeetingItem {
  final String roomName;
  final int timestamp;

  MeetingItem({required this.roomName, required this.timestamp});

  Map<String, dynamic> toJson() => {
        'roomName': roomName,
        'timestamp': timestamp,
      };

  factory MeetingItem.fromJson(Map<String, dynamic> json) => MeetingItem(
        roomName: json['roomName'] as String,
        timestamp: json['timestamp'] as int,
      );
}

class PrefsManager {
  static const _keyDisplayName = 'display_name';
  static const _keyAudioMuted = 'audio_muted';
  static const _keyVideoMuted = 'video_muted';
  static const _keyHistory = 'meeting_history';
  static const _maxHistory = 10;

  late SharedPreferences _prefs;

  Future<void> init() async {
    _prefs = await SharedPreferences.getInstance();
  }

  String get displayName => _prefs.getString(_keyDisplayName) ?? '';
  set displayName(String value) => _prefs.setString(_keyDisplayName, value);

  bool get isAudioMuted => _prefs.getBool(_keyAudioMuted) ?? false;
  set isAudioMuted(bool value) => _prefs.setBool(_keyAudioMuted, value);

  bool get isVideoMuted => _prefs.getBool(_keyVideoMuted) ?? false;
  set isVideoMuted(bool value) => _prefs.setBool(_keyVideoMuted, value);

  List<MeetingItem> getMeetingHistory() {
    final json = _prefs.getString(_keyHistory) ?? '[]';
    final list = jsonDecode(json) as List;
    return list.map((e) => MeetingItem.fromJson(e as Map<String, dynamic>)).toList();
  }

  void addMeetingToHistory(String roomName) {
    final history = getMeetingHistory();
    history.removeWhere((item) => item.roomName == roomName);
    history.insert(0, MeetingItem(
      roomName: roomName,
      timestamp: DateTime.now().millisecondsSinceEpoch,
    ));
    final trimmed = history.take(_maxHistory).toList();
    _prefs.setString(_keyHistory, jsonEncode(trimmed.map((e) => e.toJson()).toList()));
  }

  void clearHistory() {
    _prefs.remove(_keyHistory);
  }
}
